evm_rpc_client/src/lib.rs

```diff
@@ -1,5 +1,188 @@
+//! #           topics: vec![
+//! #               Hex32::from_str("0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef").unwrap(),
+//! #               Hex32::from_str("0x000000000000000000000000000000000004444c5dc75cb358380d2e3de08a90").unwrap(),
+//! #               Hex32::from_str("0x0000000000000000000000000000000aa232009084bd71a5797d089aa4edfad4").unwrap(),
+//! #           ],
+//! #           data: Hex::from_str("0x00000000000000000000000000000000000000000000000000000000cd566ae8").unwrap(),
+//! #           block_number: Some(0x161bd70_u64.into()),
+//! #           transaction_hash: Some(Hex32::from_str("0xfe5bc88d0818b66a67b0619b1b4d81bfe38029e3799c7f0eb86b33ca7dc4c811").unwrap()),
+//! #           transaction_index: Some(0x0_u64.into()),
+//! #           block_hash: Some(Hex32::from_str("0x0bbd9b12140e674cdd55e63539a25df8280a70cee3676c94d8e05fa5f868a914").unwrap()),
+//! #           log_index: Some(0x0_u64.into()),
+//! #           removed: false,
+//! #       }
+//! #   ])))
+//!     .build();
+//!
+//! let request = client.get_logs(vec![address!("0xdac17f958d2ee523a2206206994597c13d831ec7")]);
+//!
+//! let result = request
+//!     .with_cycles(10_000_000_000)
+//!     .send()
+//!     .await
+//!     .expect_consistent();
+//!
+//! assert!(result.is_ok());
+//! assert_eq!(result.unwrap().first(), Some(
+//!     &alloy_rpc_types::Log {
+//!         inner: alloy_primitives::Log {
+//!             address: address!("0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"),
+//!             data: alloy_primitives::LogData::new(
+//!                 vec![
+//!                     b256!("0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"),
+//!                     b256!("0x000000000000000000000000000000000004444c5dc75cb358380d2e3de08a90"),
+//!                     b256!("0x0000000000000000000000000000000aa232009084bd71a5797d089aa4edfad4"),
+//!                 ],
+//!                 bytes!("0x00000000000000000000000000000000000000000000000000000000cd566ae8"),
+//!             ).unwrap(),
+//!         },
+//!         block_hash: Some(b256!("0x0bbd9b12140e674cdd55e63539a25df8280a70cee3676c94d8e05fa5f868a914")),
+//!         block_number: Some(0x161bd70_u64),
+//!         block_timestamp: None,
+//!         transaction_hash: Some(b256!("0xfe5bc88d0818b66a67b0619b1b4d81bfe38029e3799c7f0eb86b33ca7dc4c811")),
+//!         transaction_index: Some(0x0_u64),
+//!         log_index: Some(0x0_u64),
+//!         removed: false,
+//!     },
+//! ));
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! ## Overriding client configuration for a specific call
+//!
+//! Besides changing the amount of cycles for a particular call as described above,
+//! it is sometimes desirable to have a custom configuration for a specific
+//! call that is different from the one used by the client for all the other calls.
+//!
+//! For example, maybe for most calls, a 2 out-of 3 strategy is good enough, but for `eth_getSlot`
```

Comment:
nit: Probably a typo
```suggestion
//! For example, maybe for most calls, a 2 out-of 3 strategy is good enough, but for `eth_getLogs`
```